### PR TITLE
BDOG-1157

### PR DESCRIPTION
**Pull request by chotai**

### CODE REVIEW

File: http-verbs-common/src/main/scala/uk/gov/hmrc/http/HttpTransport.scala [hunk 1]
```diff
@@ -101,32 +103,49 @@ trait HttpTransport
     with PostHttpTransport {}
 
 trait CoreGet {
+
   def GET[A](
```
Comment (chotai) [inline http-verbs-common/src/main/scala/uk/gov/hmrc/http/HttpTransport.scala:107]: Interestingly we provide default parameters for the other http verb calls where as here it's been overloaded. I've updated to provide default parameters for GETs too.

Comment (colin-lamed) [reply http-verbs-common/src/main/scala/uk/gov/hmrc/http/HttpTransport.scala:107]: Do we want to do the same with the new `VERB(url: URL)` functions? i.e. just one where `headers` has a default param?
Do we know if anyone is overriding the variants that we have removed?

File: http-verbs-common/src/test/scala/uk/gov/hmrc/http/RetriesSpec.scala [hunk 2]
```diff
@@ -337,7 +337,7 @@ class RetriesSpec extends AnyWordSpecLike with Matchers with MockitoSugar with S
 
       implicit val headerCarrier: HeaderCarrier = HeaderCarrier()
 
-      http.POSTForm[Option[String]](url = "doesnt-matter", Map()).futureValue shouldBe None
+      http.POSTForm[Option[String]](url = "doesnt-matter", Map.empty[String, Seq[String]], Seq.empty[(String, String)]).futureValue shouldBe None
```
Comment (chotai) [inline http-verbs-common/src/test/scala/uk/gov/hmrc/http/RetriesSpec.scala:340]: Without specifying the type `[String, Seq[String]]` on Map, the compiler isn't able to work out which of the overloaded `POSTForm` we want to use. This seems to only happen on `POSTForm`. Upgrading to this version of http-verbs can result in compiler errors so wanted to point this out.

File: http-verbs-common/src/test/scala/uk/gov/hmrc/http/HttpDeleteSpec.scala [hunk 3]
```diff
@@ -36,34 +36,96 @@ class HttpDeleteSpec extends AnyWordSpecLike with Matchers with MockitoSugar wit
...
+    val testHook1: HttpHook                         = mock[HttpHook]
+    val testHook2: HttpHook                         = mock[HttpHook]
     val hooks                                       = Seq(testHook1, testHook2)
     override val configuration: Config              = ConfigFactory.load()
     override protected val actorSystem: ActorSystem = ActorSystem("test-actor-system")
 
     def appName: String = ???
 
-    def doDelete(url: String, headers: Seq[(String, String)])(implicit hc: HeaderCarrier, ec: ExecutionContext) =
+    def doDelete(
```
Comment (colin-lamed) [inline http-verbs-common/src/test/scala/uk/gov/hmrc/http/HttpDeleteSpec.scala:48]: add `override` here?

File: http-verbs-common/src/test/scala/uk/gov/hmrc/http/RetriesSpec.scala (hunk 2 above)
Comment (colin-lamed) [reply http-verbs-common/src/test/scala/uk/gov/hmrc/http/RetriesSpec.scala:340]: I guess passing an empty body to POSTForm is unusual. If we don't have a problem with empty query params, I don't think it's an issue.

**Review by colin-lamed — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -32,7 +32,15 @@ Where `play-xx` is `play-25`, `play-26` or `play-27` depending on your version o
...
 
-Examples can be found [here](https://github.com/hmrc/http-verbs/blob/master/http-verbs-common/src/test/scala/uk/gov/hmrc/examples/Examples.scala)
+Examples can be found [here](https://github.com/hmrc/http-verbs/blob/master/http-verbs-common/src/test/scala/uk/gov/hmrc/http/examples/Examples.scala)
+
+### URLs
+
+Urls can be supplied as either java.net.URL objects or as strings (java.net.URL is recommended). A URL interpolator has been implicitly provided with http-verbs for convenience. 
+
+```
+url"http://localhost:20001/create-user"
```
Comment (colin-lamed) [inline README.md:42]: Can you include query-params and path-params in the example? This is where the value is.

File: README.md [hunk 5]
```diff
@@ -32,7 +32,15 @@ Where `play-xx` is `play-25`, `play-26` or `play-27` depending on your version o
 
 ## Usage
 
-Examples can be found [here](https://github.com/hmrc/http-verbs/blob/master/http-verbs-common/src/test/scala/uk/gov/hmrc/examples/Examples.scala)
+Examples can be found [here](https://github.com/hmrc/http-verbs/blob/master/http-verbs-common/src/test/scala/uk/gov/hmrc/http/examples/Examples.scala)
+
+### URLs
+
+Urls can be supplied as either java.net.URL objects or as strings (java.net.URL is recommended). A URL interpolator has been implicitly provided with http-verbs for convenience.
```
Comment (colin-lamed) [inline README.md:39]: Can we expand: "We recommend that you use the url interpolator for correct escaping of query and path parameters"

File: README.md (hunk 4 above)
Comment (anshulbajpai) [reply README.md:42]: I think we should link to this documentation - https://sttp.softwaremill.com/en/latest/model/uri.html. It has extensive documentation.

Comment (anshulbajpai) [reply README.md:42]: The only difference is the final type. They build their type `Uri` and we build a `java.net.URL`

**Pull request merged**